src/Symfony/Component/DependencyInjection/Compiler/CheckDefinitionValidityPass.php

```diff
@@ -58,6 +59,15 @@ public function process(ContainerBuilder $container)
                 ));
             }
 
+            // check if the service class implements the parent interface
+            if (!$definition->isAbstract() && ($class = $definition->getClass()) && ($interface = $definition->getInterface()) && !is_subclass_of($class, $interface)) {
```

Comment:
`!$definition->isAbstract()` ? when `class` attribute is defined into abstract parent service, we should validate it too?